### PR TITLE
[CHORE]Validate job applicant doc, before creation of job offer

### PR DESCRIPTION
**Pull request by NoBoneZ**
## Is this a Feature, Chore or Bug?
- [] Feature
- [x] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
Validation of job applicant , before the creation of job offer, to avoid errors(caused by empty mandatory fields) popping up when employee is about to created 

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [x] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.
Job Offer

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [x] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [x] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [x] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [x] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/hiring/utils.py [hunk 1]
```diff
@@ -35,6 +36,8 @@ def get_performance_profile_guid():
         return get_url(file_path)
 
 def validate_job_offer(doc, method):
+    job_applicant = frappe.get_doc("Job Applicant", doc.job_applicant)
```
Comment (pjamsheer) [inline one_fm/hiring/utils.py:39]: We are validating the Job Applicant while we set Selected the field `one_fm_applicant_status`. So in job offer we need to validate only the value of `one_fm_applicant_status` in the Job Applicant.
1. get the `one_fm_applicant_status` from Job Applicant
2. check if the value equals to `Selected`, then proceed
3. check if the value not equals to `Selected`, then tell the user that we can create job offer for selected candidate only

Comment (NoBoneZ) [reply one_fm/hiring/utils.py:39]: @pjamsheer okay, i could add that, as further validation, but the issue was that, the absence of data in some mandatory fields of the job applicant was causing an issue, and due to the "save as draft" story we worked on for the job applicant, there is a doc.flag.ignore_validate, to make it save. Removing that(doc.flag) would solve the issue, but the "save as draft" functionality we added will not work cos of the validation

Comment (pjamsheer) [reply one_fm/hiring/utils.py:39]: @NoBoneZ So selected applicant is going to fill the applicant doc data or after fill the applicant doc data we are selecting the applicant

Comment (NoBoneZ) [reply one_fm/hiring/utils.py:39]: oh, okay okay, i will add the validation to it

**Comment by mymi14s** [comment]
@pjamsheer @NoBoneZ is this good to go?

**Comment by NoBoneZ** [comment]
> @pjamsheer @NoBoneZ is this good to go?

@mymi14s ,yes sir, what Mr jam spoke was already being in effect, cos selecting an applicant automatically creates a job offer, so I think it is OK sir

**Review by mymi14s — APPROVED** [review]

**Pull request merged**